src/shadowbox/server/server_access_key.ts

```diff
@@ -181,19 +179,28 @@ export class ServerAccessKeyRepository implements AccessKeyRepository {
   }
 
   get dataLimit(): DataLimit|undefined {
-    return this.accessKeyDataLimit;
+    return this.defaultDataLimit;
   }
 
-  setAccessKeyDataLimit(limit: DataLimit): Promise<void> {
```

Comment:
This validation was removed because it duplicates validation done at the API boundary.  By the time we're here, we don't have an unknown JSON request but a known valid DataLimit.